lib/experimental/fonts/font-library/class-wp-font-library.php

```diff
@@ -19,12 +19,17 @@
  * @since 6.4.0
  */
 class WP_Font_Library {
-
+	/*
+	* As of PHP 8.1.12, which includes libmagic/file update to version 5.42,
+	* the expected mime type for WOFF files is 'font/woff'.
+	*
+	* See https://github.com/php/php-src/issues/8805.
+	*/
 	const ALLOWED_FONT_MIME_TYPES = array(
 		'otf'   => 'font/otf',
-		'ttf'   => 'font/ttf',
-		'woff'  => 'font/woff',
-		'woff2' => 'font/woff2',
+		'ttf'   => PHP_VERSION_ID >= 80112 ? 'font/ttf' : 'application/x-font-ttf',
+		'woff'  => PHP_VERSION_ID >= 80112 ? 'font/woff' : 'application/font-woff',
+		'woff2' => PHP_VERSION_ID >= 80112 ? 'font/woff2' : 'application/font-woff2',
```

Comment:
I'm not sure about checking for a specific PHP version here; I can't see an example of this anywhere else in the codebase. I'm wondering if we could create an array of allowed mime types to loop through, rather than using the `PHP_VERSION_ID` check. Something along the lines of:

```suggestion
		'ttf'   => ['font/ttf', 'application/x-font-ttf'],
		'woff'  => ['font/woff', 'application/font-woff'],
		'woff2' => ['font/woff2', 'application/font-woff2'],
```

Reply:
Yeah. I agree it is not a good approach. But the mime check doesn't support multiple values in core. So, we should proceed with this approach until there is support for multiple values.